interface/src/app/analyze/analysis/analysis.js

```diff
@@ -187,21 +221,20 @@ AnnotationType, SampleBin) {
     ]
   };
 
-  // TODO these exampleCols are temporarily hard-coded until a column chooser
-  // feature can be added
-  $scope.analysis = {
-    exampleCols: [
-      {'typename': 'genotype'},
-      {'typename': 'medium'},
-      {'typename': 'strain'}
-    ]
-  };
-
   // populate sample details
   // FIXME implement this loop as a method on SampleBin? (prob. with caching)
+  $scope.analysis.status = 'Retrieving sample details';
+  var pArrSamples = [];
   for (var i = 0; i < SampleBin.heatmapData.samples.length; i++) {
-    SampleBin.getSampleDetails(SampleBin.heatmapData.samples[i]);
+    pArrSamples.push(
+      SampleBin.getSampleDetails(SampleBin.heatmapData.samples[i])
+    );
   }
+  $q.all(pArrSamples).then(function() {
```

Comment:
Whoops. I should have looked more closely at the context here... The relevant API is `Sample`, not `Activity`. But the logic I'm getting at is the same. Caching is not yet implemented for this `Sample` data, but once I have generalized that code, the UI will cache this sample data as well as the activity data so the same logic applies.